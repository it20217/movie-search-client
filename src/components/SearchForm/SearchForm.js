import {useState} from 'react';
import { useNavigate } from "react-router-dom";
import styles from './SearchForm.module.scss';

function SearchForm(props) {
  const [query, setQuery] = useState('');
  const navigate = useNavigate();

  const handleInput = (e) => {
    e.preventDefault();
    setQuery(e.target.value);
  }
  const handleFind = (e) => {
    navigate(`/search/${query}`);
    setQuery('');
  }
  console.log(props.marginTop)
  return(
    <div className={styles.search}>
      <div 
        className={styles["form-wrapper"]}
        style={{marginTop: props.marginTop}}
      >
        <div className={styles["search-input"]}>
          <input 
            type="search" 
            value={query}
            className={styles["search-input"]} 
            placeholder={"Spiderman: No Way Home"}
            onChange={handleInput}
          />
        </div>
        <div className={styles["btn-container"]}>
          <button 
            type="button"
            className={styles["form-btn"]}
            onClick={handleFind}
          >
            Find
          </button>
        </div>
      </div>
    </div>
  )
}

export default SearchForm;