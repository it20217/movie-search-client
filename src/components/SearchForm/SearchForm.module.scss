$gold: #ecdb20;
$black: #273461;

.search {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 100%;
  margin: 0 20px;
  white-space: nowrap;
}
.margin-top {
  margin-top: 80px;
}

.form-wrapper {
  display: flex;
  flex-direction: row;
  white-space: nowrap;
  width: 100%;
  max-width: 550px;
}
.search-input {
  flex: 2;
  padding-left: 10px;
  font-size: 16px;
  >input {
    font-family: Montserrat;
    width: 100%;
    height: 48px;
    font-size: 18px;
    color: $black;
    border: 3px solid #273461;
    border-radius: 4px;
    outline: none;
  }
  >input::placeholder {
    font-style: italic;
    font-size: 18px;
  }
  //remove 'x' from input field
  > input[type="search"]::-webkit-search-decoration,
    input[type="search"]::-webkit-search-cancel-button,
    input[type="search"]::-webkit-search-results-button,
    input[type="search"]::-webkit-search-results-decoration {
    display: none;
  }
}
.btn-container {
  min-width: 100px;
}
.form-btn {
  font-family: Montserrat;
  margin: 0 10px;
  padding: 0 32px;
  height: 48px;
  border:3px solid #273461 ;
  border-radius: 6px;
  //background-color: $gold;
  //background-image: linear-gradient(to right,#f05a28,#e80a89);
  font-size: 24px;
  line-height: initial;
}
