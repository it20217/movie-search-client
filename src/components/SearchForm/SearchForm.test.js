import React from 'react';
/** Snapshots rendering */
import renderer from "react-test-renderer";
/** Component */
import SearchForm from './SearchForm';
/** React router */
import { BrowserRouter as Router } from 'react-router-dom';

describe("SearchForm page component", () => {
  it("renders correctly", () => {
    const component = renderer.create(<Router><SearchForm/></Router>).toJSON();
    expect(component).toMatchSnapshot();
  })
})