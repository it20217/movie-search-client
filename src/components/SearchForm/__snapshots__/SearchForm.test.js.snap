// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm page component renders correctly 1`] = `
<div
  className="search"
>
  <div
    className="form-wrapper"
    style={
      Object {
        "marginTop": undefined,
      }
    }
  >
    <div
      className="search-input"
    >
      <input
        className="search-input"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Spiderman: No Way Home"
        style={
          Object {
            "borderColor": "#273461",
          }
        }
        type="search"
        value=""
      />
    </div>
    <div
      className="btn-container"
    >
      <button
        className="form-btn"
        onClick={[Function]}
        style={
          Object {
            "Color": "#273461",
            "borderColor": "#273461",
          }
        }
        type="button"
      >
        <svg
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M14 5l7 7m0 0l-7 7m7-7H3"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
